src/linear_algebra/finite_dimensional.lean

```diff
@@ -1737,6 +1741,21 @@ begin
   exact subalgebra.eq_bot_of_finrank_one h,
 end
 
+lemma subalgebra.eq_bot_or_top_of_finrank (S : subalgebra F E) (hr : finrank F E = 2) :
+  S = ⊥ ∨ S = ⊤ :=
```

Comment:
I think this can be stated with `is_simple`

Reply:
I can probably prove the following statement: 
```lean 
lemma subalgebra.eq_bot_or_top_of_finrank (S : subalgebra F E) (hr : finrank F E = 2) :
  is_simple_module F S
``` 
but then I'll need to import `ring_theory.simple_module` into `linear_algebra.finite_dimensional`. I guess that's probably ok...

Reply:
`finrank F E ≤ 2` would be more general and not harder to use since there is `eq.le`.

Reply:
Yeah, that's what I mean.

Reply:
> `finrank F E ≤ 2` would be more general and not harder to use since there is `eq.le`.

But, then I would need to add the hypothesis `finite_dimensional F E` to exclude the case `finrank F E = 0`? That would make the result slighter harder to use and I don't think there is much gain since there is already `subalgebra.eq_bot_of_finrank_one` to deal with the rank one case.

Reply:
> I think @eric-wieser means [`is_simple_order (subalgebra F E)`](https://leanprover-community.github.io/mathlib_docs/order/atoms.html#is_simple_order)? The image of `F` in `E` is a `F`-submodule that is neither bot nor top, right?

I see, I guess it gives a more precise statement that the one there is now..

Reply:
> But, then I would need to add the hypothesis finite_dimensional F E to exclude the case finrank F E = 0?

> I guess it gives a more precise statement that the one there is now..

I think it's exactly the same except `is_simple_order` also requires nontriviality? Because it requires nontriviality, `finrank F E ≤ 2` no longer works (even barring the infinite rank issue), so yes, let's just stick with `finrank F E = 2`.

Reply:
I added `subalgebra.bot_eq_top_iff_dim_eq_one` and used it to prove 
```lean
def subalgebra.is_simple_order_of_finrank (hr : finrank F E = 2) :
  is_simple_order (subalgebra F E)
```
There is still probably some room for improvement in that proof though...